exporters/otlp/otlpmetric/internal/oconf/options.go

```diff
@@ -263,11 +265,32 @@ func NewGRPCOption(fn func(cfg Config) Config) GRPCOption {
 
 func WithEndpoint(endpoint string) GenericOption {
 	return newGenericOption(func(cfg Config) Config {
-		cfg.Metrics.Endpoint = endpoint
+		// Add scheme if not present
+		if !internal.HasScheme(endpoint) {
+			endpoint = getScheme(cfg) + "://" + endpoint
+		}
+		u, err := url.Parse(endpoint)
+		if err != nil {
+			global.Error(err, "parse url", "input", endpoint)
+			return cfg
+		}
+		cfg.Metrics.Endpoint = u.Host
+		// For OTLP/HTTP endpoint URLs without a per-signal
+		// configuration, the passed endpoint is used as a base URL
+		// and the signals are sent to these paths relative to that.
+		cfg.Metrics.URLPath = path.Join(u.Path, DefaultMetricsPath)
 		return cfg
 	})
 }
 
+// getScheme fetches scheme based on config.
+func getScheme(cfg Config) string {
```

Comment:
I'm not sure this is a simple binary choice nor is it clear that `Insecure` necessarily translates to `http`.

Reply:
Actually, I have referred that function from https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/otlp/otlptrace/otlptracehttp/client.go#L320

Reply:
Hey @Aneurysm9, Could you please take a look at my comment?

Reply:
Not sure what happened to the link I attempted to put in that comment.  See https://github.com/grpc/grpc/blob/master/doc/naming.md.

Reply:
@Aneurysm9 It looks OK. See https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/protocol/exporter.md


>  A scheme of `https` indicates a secure connection and takes precedence over the insecure configuration setting. A scheme of `http` indicates an insecure connection and takes precedence over the insecure configuration setting. If the gRPC client implementation does not support an endpoint with a scheme of http or https then the endpoint SHOULD be transformed to the most sensible format for that implementation.